### PR TITLE
Fix missing catch

### PR DESCRIPTION
**Pull request by Nimphal**
### What changed
The locations and move controllers were updated to wrap calls to the API in try catch blocks.

### Why did it change
If the API returned an error, we weren't handling it. It would be logged, but it wouldn't exit the req cycle and the users could end up seeing partial data.

#### Automated testing

- [x] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/main/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics

### CODE REVIEW

File: app/locations/controllers.js [hunk 1]
```diff
@@ -4,39 +4,37 @@ async function locations(req, res, next) {
...
+
+      // The locations have been uniqued based on title to prevent
+      // duplicates when multiple suppliers have the same location
+      userLocations = uniqBy(supplierLocations.flat(), 'id')
+    }
+  } catch (error) {
+    return next(error)
   }
 
   req.session.user.locations = userLocations
```
Comment (teneightfive) [inline app/locations/controllers.js:40]: It won't let me comment on the line but lines 42 to 46 were the ones removed by my PR so be good to just double check they don't creep back in with this. Will likely be resolved with a rebase :)

File: app/locations/controllers.test.js [hunk 2]
```diff
@@ -214,5 +204,25 @@ describe('Locations controllers', function () {
...
       })
     })
+
+    context('when there is an API error', function () {
+      it('should fail gracefully for regions API', async function () {
+        req.session.user.permissions = ['allocation:create']
+        req.services.referenceData.getRegions = sinon.fake.returns(
+          Promise.reject(new Error())
+        )
+        await proxiedController.locations(req, res, nextSpy)
```
Comment (teneightfive) [inline app/locations/controllers.test.js:214]: I also made a change to remove the need to use proxyquire here as we weren't mocking any dependencies of this module so it was unnecessary. You can change this to the following once rebase:
```suggestion
        await locationsController(req, res, nextSpy)
```

**Review by teneightfive — APPROVED** [review]
Changes look good. Might just want to double check a few once rebase on top of #1567

File: app/locations/controllers.js (hunk 1 above)
Comment (Nimphal) [reply app/locations/controllers.js:40]: Git was clever enough to not make a fuss about this, so all is as it should be :)

**Pull request merged**